server/src/test/java/org/opensearch/search/CreatePitSingleNodeTests.java

```diff
@@ -223,7 +228,7 @@ public void testSearchWithFirstPhaseKeepAliveExpiry() throws ExecutionException,
         // since first phase temporary keep alive is set at 1 second in this test file
         // and create pit request keep alive is less than that, keep alive is set to 1 second, (max of 2 keep alives)
         // so reader context will clear up after 1 second
-        Thread.sleep(1000);
+        Thread.sleep(1200);
```

Comment:
`Thread.sleep()` calls make a test flaky. As everything depends on the state of the server at the time tests are running and CPU context switch that is happening, the code that we expect to finish in 1200ms may take more than that. Is there any way to provide listener?